doc/arch/adr-003-data-retention.md

```diff
@@ -0,0 +1,287 @@
+the constraints of GDPR by not storing personally identifiable data for
+longer than is needed.
+
+The goal being that we can complete this prior to the change to Notify so that
+we avoid a data black hole where we do not have the data to answer common
+data queries.
+
+## Specifics
+
+### Common data use scenarios
+
+Based on a chat with Paul (our performance analyst) we were demonstrated to a
+number of scenarios of data look ups and usage. As this was only a short session
+this list is likely incomplete.
+
+#### Data that is currently looked up
+
+- Number of people subscribed to list
+  - Given a date range: how many people subscribed in that range; how many
+    unscribed; total number of subscribers during that time period.
+- Activity Level of list
+  - Given a date range: how many emails were sent out to that list; how many
+    bulletins a list has associated with it; how many emails a list has
+    genertaed; whether lists have never produced an email.
+- Overall totals
+  - Counts of how many emails we have sent over a specific period of time or
+    total (mostly used internally)
+
+#### Support queries
+
+- People have been unsubscribed and don’t know why (Govdelivery tracks a user
+  unsubscribed but not which lists)
+- Why a user was unsubscribed - this can happen automatically or a user
+  performs the action.
+- Occasional need to check email. Reason this can happen is that we get a query
+  because the link may not be working, so we would look at what was sent.
+
+#### Things that would be useful
+
+- Knowing difference between users who subscribed by themselves or were
+  imported (our initial data import will leave us with inconsistent stats)
+- Ability to recreate subscriptions when people have unsubscribed.
+
+### Suggested Changes
+
+This is a summary of a collection of changes to the data architecture to enable
+most statistical data to be retained at a relatively low level of pain of
+implementation and should enable future decisions to be made about usage and
+storage with a relatively full set of data.
+
+#### Email and DeliveryAttempt
+
+The largest source of data is the Email table which has a relationship with
+DeliveryAttempt. This is the most pressing problem to resolve in terms of
+data retention as this quickly builds up a substantial quantity of data.
+
+The suggested approach to dealing with this is to have two separate
+processes which can both be adjusted as we learn more about the system.
+
+The first of these processes is an archiving one which should happen within
+24 hours of an email reaching a final state (either sent or failed). When this
+has occurred the row should be marked that as archived (suggestion is for a
+`archived_at` timestamp) so it not as archived again and is known to be safe
+to delete later.
+
+The archiving would not involve any personally identifiable data and would
+only include the subject field from the email. As an example:
+
+```
```

Comment:
Small thing, but might be nice to mark this as JSON for the syntax highlighting... or maybe the comments make it look ugly?

Reply:
Yeah just tried it and it looked minging sadly. It always makes me sad how JSON doesn't allow comments. I will however provide a rendered link to this ADR in the description